### DISCUSSION TITLE
s3api changed in v1.0?

### DISCUSSION THREAD

**Discussion by cjbd** [duckdb/duckdb#12419] [Q&A]
Hello, I started using `duckdb` on windows 11 recently, I was using `v0.10`, and I was able query s3 like [this document](https://duckdb.org/docs/extensions/httpfs/s3api)

```sql
SELECT *
FROM 's3://my-bucket/file.csv';
```

yesterday, I downloaded latest `v1.0` version and above sql stopped working, I need to wrap the s3 url in `read_csv`, otherwise there's error message:

```
Catalog Error: Table with name s3://my-bucket/file.csv does not exist!
Did you mean "pg_database"?
LINE 1: select * from 's3://my-bucket/file.csv'
```

is this an issue or we have to wrap s3 url inside read_csv function from now on? 
thank you!

**Comment by carlopi** [comment]
This looks like httpfs can't be autoloaded.
Can you check whether `SELECT current_setting('autoload_known_extensions')` is true or false?
And same for `SELECT current_setting('autoinstall_known_extensions')`?

If both are true, the statement should work properly.
Otherwise you might need to explicitly do `INSTALL httpfs` (once) and `LOAD httpfs` (once per session).

**Comment by cjbd** [comment]
@carlopi , thank you for your fast reply, here's my test on windows 11:

```
v1.0.0 1f98600c2c
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
D SELECT current_setting('autoload_known_extensions');
┌──────────────────────────────────────────────┐
│ current_setting('autoload_known_extensions') │
│                   boolean                    │
├──────────────────────────────────────────────┤
│ true                                         │
└──────────────────────────────────────────────┘
D SELECT current_setting('autoinstall_known_extensions');
┌─────────────────────────────────────────────────┐
│ current_setting('autoinstall_known_extensions') │
│                     boolean                     │
├─────────────────────────────────────────────────┤
│ true                                            │
└─────────────────────────────────────────────────┘
D INSTALL httpfs;
D LOAD httpfs;

D select * from 's3://my-bucket/file.csv';
Catalog Error: Table with name s3://my-bucket/file.csv does not exist!
Did you mean "pg_database"?
LINE 1: select * from 's3://my-bucket/file.csv';
                      ^
```

I removed the statements to set credentials and changed bucket name, rest are from console output

**Reply by carlopi** [reply]
If I try locally I get:
```
carlo@ScroogeMcDuck duckdb % duckdb
v1.0.0 1f98600c2c
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
D SELECT *
  FROM 's3://my-bucket/file.csv';
HTTP Error: HTTP GET error on 'https://my-bucket.s3.amazonaws.com/file.csv' (HTTP 403)
```
That is the output I will expect. I am a bit puzzled, will need to summon someone else.

**Comment by cjbd** [comment]
hello, anyone knows which part of code controls this behavior, is this what document says about [replacement scan](https://duckdb.org/docs/api/c/replacement_scans.html)?
maybe i can help to fix it
